### ISSUE TITLE
Non parsing errors silently fails and causes timeout. No error displayed.

### ISSUE THREAD

**Issue by JonWallsten** [mucsi96/nightwatch-cucumber#151]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report
[x ] feature request
[ ] support request => Please do not submit support request here, instead use https://groups.google.com/d/forum/nightwatch-cucumber
```

**Current behavior**
When a type error or similar occur in a custom command, step definition or assertion the test will silently fail and cause a timeout.

**Expected/desired behavior**
I should get an error message and stack trace to be able to find the issue.

**Reproduction of the problem**
If the current behavior is a bug or you can illustrate your feature request better with an example, please provide the steps to reproduce and if possible a minimal demo of the problem via https://c9.io, http://www.koding.com, https://www.nitrous.io or similar.

```
#testCommand.js
const testCommand = function() {
    var test = undefinedVar;
    return this;
};
exports.command = testCommand;

#stepDef.js
this.Given(/^Test command$/, function(client) {
    client.testCommand();
});

#test.feature
Scenario: Test command
    Given Test command
```

**What is the expected behavior?**
It should fail and output an error

**What is the motivation / use case for changing the behavior?**
It's right now really hard to find the cause of the issue. Only way is to console log you way through the file and see how far you get.

**Please tell us about your environment:**

* **nightwatch-cucumber version:**
5.6.3
* **nightwatch version:**
0.9.9
* **cucumber version:**
1.3.0
* **Node.js version:**
7.2.0

**Comment by mucsi96** [comment]
Hi @JonWallsten! Thanks for feedback! Can you provide a pull request with a solution?

**Comment by JonWallsten** [comment]
Honestly I don't even know where to start. I did a quick test with Nightwatch only. It handles type errors as well and gives an error message. So where is it swallowed? And why doesn't it break the step? If you have any ideas I can start poking around.

**Comment by JonWallsten** [comment]
This is what is looks like when I disconnect Nightwatch-cucumber and try to run the following test

```
module.exports = {
    "Demo test Google": function(browser) {
        browser
            .openPage()
            .end();
    }
};
```

Inside I have deliberately added this: 
```
var foo = test; // Test is undefined
```

And the result in Nightwatch
```
Starting selenium server... started - PID:  1120

[Test] Test Suite
=====================

Running:  Demo test Google
Error while running openPage command: test is not defined
    at Object.Demo test Google (C:\my-path\temp\test.js:4:14)
    at Module.call (C:\my-path\node_modules\nightwatch\lib\runner\module.js:62:34)
    at C:\my-path\node_modules\nightwatch\lib\runner\testcase.js:70:29
    at _fulfilled (C:\my-path\node_modules\q\q.js:834:54)
    at self.promiseDispatch.done (C:\my-path\node_modules\q\q.js:863:30)
    at Promise.promise.promiseDispatch (C:\my-path\node_modules\q\q.js:796:13)
    at C:\my-path\node_modules\q\q.js:556:49
    at runSingle (C:\my-path\node_modules\q\q.js:137:13)
    at flush (C:\my-path\node_modules\q\q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)

FAILED:  1 errors (6.147s)

 _________________________________________________

 TEST FAILURE: 1 error during execution,  0 assertions failed, 0 passed. (6.474s)

 × test

   - Demo test Google (6.147s)

   Error while running openPage command: test is not defined
       at Object.Demo test Google (C:\my-path\temp\test.js:4:14)
       at _combinedTickCallback (internal/process/next_tick.js:67:7)


Error while running openPage command: test is not defined
    at Object.Demo test Google (C:\my-path\temp\test.js:4:14)
    at Module.call (C:\my-path\node_modules\nightwatch\lib\runner\module.js:62:34)
    at C:\my-path\node_modules\nightwatch\lib\runner\testcase.js:70:29
    at _fulfilled (C:\my-path\node_modules\q\q.js:834:54)
    at self.promiseDispatch.done (C:\my-path\node_modules\q\q.js:863:30)
    at Promise.promise.promiseDispatch (C:\my-path\node_modules\q\q.js:796:13)
    at C:\my-path\node_modules\q\q.js:556:49
    at runSingle (C:\my-path\node_modules\q\q.js:137:13)
    at flush (C:\my-path\node_modules\q\q.js:125:13)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)

npm ERR! Windows_NT 10.0.14393
npm ERR! argv "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\bin\\npm-cli.js" "run" "start"
npm ERR! node v7.2.0
npm ERR! npm  v3.10.9
npm ERR! code ELIFECYCLE
npm ERR! @company/ui-test@0.2.0 start: `node src/index.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the @company/ui-test@0.2.0 start script 'node src/index.js'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the @company/ui-test package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node src/index.js
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs @company/ui-test
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls @company/ui-test
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     C:\my-path\npm-debug.log
```

**Comment by JonWallsten** [comment]
Can the problem be Cucumber? Since Nightwatch is handling the errors fine when you don't use steps.

**Comment by JonWallsten** [comment]
@mucsi96 Ping

**Comment by JonWallsten** [comment]
Yeah, I would not have figured that out... ;) Thank you!

**Comment by mucsi96** [comment]
Yeah :) You are welcome! Thanks for reporting! And sorry for so long time to fix this

**Comment by JonWallsten** [comment]
Works perfectly. Will try to contribute whenever I can. But it's really complex stuff when you're out of context.

**Comment by JonWallsten** [comment]
@mucsi96 Almost perfect! ;)
```
TypeError: Cannot read property 'indexOf' of undefined
    at Nightwatch.client.get.terminate (C:\my-path\node_modules\nightwatch-cucumber\lib\nightwatch-api.js:169:35)
    at Nightwatch.assert [as assertion] (C:\my-path\node_modules\nightwatch\lib\core\assertion.js:262:14)
    at Array.<anonymous> (C:\my-path\node_modules\nightwatch\lib\core\assertion.js:106:19)
    at triggerDataUpdate (C:\my-path\node_modules\node-resemble-v2\node_modules\resemblejs\resemble.js:97:28)
    at onceWeHaveBoth (C:\my-path\node_modules\node-resemble-v2\node_modules\resemblejs\resemble.js:583:6)
    at hiddenImage.onload (C:\my-path\node_modules\node-resemble-v2\node_modules\resemblejs\resemble.js:194:5)
    at Image.src (C:\my-path\node_modules\canvas\lib\image.js:30:17)
    at loadImageData (C:\my-path\node_modules\node-resemble-v2\node_modules\resemblejs\resemble.js:198:21)
    at compare (C:\my-path\node_modules\node-resemble-v2\node_modules\resemblejs\resemble.js:589:4)
    at wrapper (C:\my-path\node_modules\node-resemble-v2\node_modules\resemblejs\resemble.js:681:7)
```

I will investigate and do PR.

**Comment by mucsi96** [comment]
Yes :) I also noticed it. Fixed. Thanks!

**Comment by mucsi96** [comment]
Released in `5.6.6`

**Comment by JonWallsten** [comment]
Perfect! Thanks!
